# valid-config

üíº This rule is enabled in the following configs: ‚úîÔ∏è `legacy-recommended`, ‚úÖ `recommended`.

<!-- end auto-generated rule header -->

The rule checks that, if present, the `config` property is an object.

Example of **incorrect** code for this rule:

```json
{
	"config": true
}
```

Example of **correct** code for this rule:

```json
{
	"config": {
		"name": "foo",
		"config": {
			"port": "8080"
		}
	}
}
```
