# valid-directories

ğŸ’¼ This rule is enabled in the following configs: âœ”ï¸ `legacy-recommended`, âœ… `recommended`.

<!-- end auto-generated rule header -->

The rule checks that, if present, the `directories` property is an object.

Example of **incorrect** code for this rule:

```json
{
	"directories": true
}
```

Example of **correct** code for this rule:

```json
{
	"directories": {
		"bin": "dist/bin",
		"man": "docs"
	}
}
```
