# valid-homepage

ğŸ’¼ This rule is enabled in the following configs: âœ”ï¸ `legacy-recommended`, âœ… `recommended`, ğŸ“¦ `recommended-publishable`.

<!-- end auto-generated rule header -->

This rule does the following checks on the value of the `homepage` property:

- It must be a `string`
- The value should be a valid URL

Example of **incorrect** code for this rule:

```json
{
	"homepage": "trent@nin.com"
}
```

Example of **correct** code for this rule:

```json
{
	"homepage": "https://nin.com"
}
```
